runtime/mux.go

```diff
@@ -164,6 +166,17 @@ func (s *ServeMux) Handle(meth string, pat Pattern, h HandlerFunc) {
 	s.handlers[meth] = append([]handler{{pat: pat, h: h}}, s.handlers[meth]...)
 }
 
+// HandlePath Handle path pattern form outside
+func (s *ServeMux) HandlePath(meth string, pathPattern string, h HandlerFunc) {
+	compiler, err := httprule.Parse(pathPattern)
+	if err != nil {
+		grpclog.Fatalf("Parse path to compiler failed: %v", err)
+	}
+	tp := compiler.Compile()
+	pattern := MustPattern(NewPattern(tp.Version, tp.OpCodes, tp.Pool, tp.Verb))
+	s.Handle(meth, pattern, h)
+}
```

Comment:
Lets make this fail instead of panic if the pattern is invalid
```suggestion
func (s *ServeMux) HandlePath(meth string, pathPattern string, h HandlerFunc) error {
	compiler, err := httprule.Parse(pathPattern)
	if err != nil {
		return fmt.Errorf("parsing path pattern: %w", err)
	}
	tp := compiler.Compile()
	pattern, err := NewPattern(tp.Version, tp.OpCodes, tp.Pool, tp.Verb)
	if err != nil {
		return fmt.Errorf("creating new pattern: %w", err)
	}
	s.Handle(meth, pattern, h)
}
```